core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -97,14 +106,15 @@ public static boolean isIngestNode(Settings settings) {
      * and updated.
      * </p>
      *
-     * @param nodeId     the nodes unique id.
-     * @param address    the nodes transport address
-     * @param attributes node attributes
-     * @param roles      node roles
-     * @param version    the version of the node.
+     * @param id               the nodes unique (ephemeral and persistent) node id
```

Comment:
I found it very cumbersome to generates those uuids in all the tests. I think that if you don't specify an ephemeral id, it's safe to assume you want a new node? if you don't and you need equality you will find out soon enough?